test/viewmodels/comment_view_model_integration_test.dart

```diff
@@ -0,0 +1,149 @@
+    late CommentRepositoryService commentRepository;
+    late AutoDisposeFamilyAsyncNotifierProvider<CommentViewModel,
+        List<CommentPost>, PostIdFirestore> commentViewModelProvider;
+    late ProviderContainer container;
+
+    late PostIdFirestore postId;
+
+    setUp(() async {
+      fakeFirestore = FakeFirebaseFirestore();
+      commentDataGenerator = CommentDataGenerator();
+
+      container = ProviderContainer(
+        overrides: [
+          firestoreProvider.overrideWithValue(fakeFirestore),
+        ],
+      );
+
+      userRepository = container.read(userRepositoryProvider);
+      commentRepository = container.read(commentRepositoryProvider);
+
+      final post = FirestorePostGenerator().generatePostAt(userPosition0);
+      await setPostFirestore(post, fakeFirestore);
+      postId = post.id;
+
+      commentViewModelProvider = commentListProvider(postId);
+    });
+
+    group("display of comments", () {
+      test("No comments are returned when the database is empty", () async {
+        final comments = await container.read(commentViewModelProvider.future);
+
+        expect(comments, isEmpty);
+      });
+
+      test("Multiple comments are returned when the database has comments",
+          () async {
+        const numberOwners = 3;
+        const numberComments = 19;
+
+        // Create and add the comments owner to the database
+        final owners =
+            FirestoreUserGenerator.generateUserFirestore(numberOwners);
+        for (final owner in owners) {
+          await userRepository.setUser(owner.uid, owner.data);
+        }
+
+        // Create and add the comments to the database
+        final commentsData = commentDataGenerator
+            .generateCommentData(numberComments)
+            .mapIndexed(
+              (index, commentData) => CommentData(
+                ownerId: owners[index % numberOwners].uid,
+                publicationTime: commentData.publicationTime,
+                voteScore: commentData.voteScore,
+                content: commentData.content,
+              ),
+            )
+            .toList();
+
+        final expectedComments = <CommentPost>[];
+
+        for (var i = 0; i < numberComments; i++) {
+          final commentData = commentsData[i];
+          final owner = owners[i % numberOwners];
+
+          await commentRepository.addComment(
+            postId,
+            commentData,
+          );
+
+          final comment = CommentPost.from(commentData, owner.data);
+
+          expectedComments.add(comment);
+        }
+
+        final actualComments =
+            await container.read(commentViewModelProvider.future);
+
+        // Check that all the comments are present
+        expect(actualComments, unorderedEquals(expectedComments));
+
+        final expectedSortedComments = expectedComments.sorted(
+          (commentA, commentB) =>
+              -commentA.publicationDate.compareTo(commentB.publicationDate),
+        );
```

Comment:
I'm not a big fan of this test, since this is just a re-implementation of your code. I think that a more interesting test would be to check the intuitive order, i.e. that, when iterating over all posts, the post at index i should have been published after the one at index i+1.

Reply:
That is a good point indeed :+1: 
I have refactored the test to check the intuitive ordering in ca7e18d983cb1757ab20c83df6bc5cca88fa8021.
Thanks for the suggestion !